pkg/state/options.go

```diff
@@ -183,3 +184,41 @@ func WithKindTailEvents(n int) WatchKindOption {
 		opts.TailEvents = n
 	}
 }
+
+// WatchWithLabelQuery appends a label query to the list options.
+func WatchWithLabelQuery(query resource.LabelQuery) WatchKindOption {
```

Comment:
I wish we could reuse the same methods for list and watch as now the API will look a bit unobvious.
There's `WithLabelQuery` which only works with the list requests. 
Or we should rename `WithLabelQuery` to `ListWithLabelQuery`.

Or another idea is to have subfolder `watch` to make it be `watch.WithLabelQuery`.

Reply:
it's still possible to do that bypassing `With` methods, I will think about better option tomorrow... I don't like it either